packages/components/src/components/select-native/select-native.jsx

```diff
@@ -4,109 +4,107 @@ import PropTypes from 'prop-types';
 import Field from '../field';
 import Icon from '../icon/icon.jsx';
 
-class SelectNative extends React.Component {
-    getDisplayText = value => {
-        const { list_items } = this.props;
-        const dropdown_items = Array.isArray(list_items) ? list_items : [].concat(...Object.values(list_items));
-        const list_obj = dropdown_items.find(item => {
-            if (typeof item.value !== 'string') return item.value === value;
-            return item.value.toLowerCase() === value.toLowerCase();
-        });
+const getDisplayText = (list_items, value) => {
+    const dropdown_items = Array.isArray(list_items) ? list_items : [].concat(...Object.values(list_items));
+    const list_obj = dropdown_items.find(item =>
+        typeof item.value !== 'string' ? item.value === value : item.value.toLowerCase() === value.toLowerCase()
+    );
 
-        if (list_obj) return list_obj.text;
-        return '';
-    };
-    render() {
-        const {
-            className,
-            classNameDisplay,
-            list_items,
-            value,
-            label,
-            placeholder,
-            use_text,
-            error,
-            hint,
-            disabled,
-            should_show_empty_option = true,
-            ...props
-        } = this.props;
-        return (
+    if (list_obj) return list_obj.text;
+    return '';
+};
+
+const SelectNative = ({
+    className,
+    classNameDisplay,
+    disabled,
+    error,
+    hint,
+    label,
+    list_items,
+    placeholder,
+    should_show_empty_option = true,
+    use_text,
+    value,
+    ...props
+}) => (
+    <div
+        className={classNames(className, 'dc-select-native', {
+            'dc-select-native--disabled': disabled,
+            'dc-select-native--error': error,
+        })}
+    >
+        <div className='dc-select-native__wrapper'>
             <div
-                className={classNames(className, 'dc-select-native', {
-                    'dc-select-native--disabled': disabled,
-                    'dc-select-native--error': error,
+                className={classNames('dc-input', {
+                    'dc-input__disabled': disabled,
```

Comment:
Disabled is a modified state of the element. I suggest keeping the `--disabled` in order to comply with BEM.

Reply:
Totally agree. Not related to my changes but will fix it here. (Since I also improved some other parts not related to this PR, this is a good punishment :smile: )